server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -238,45 +250,129 @@ def aggregate(
         step: "AggregateStep",
     ) -> StepContext:
         agg_selected: list[Field] = []
+        window_selected: list[tuple[int, Field]] = []
+        window_subquery_list: list[Tables] = []
+
+        def _build_window_subquery() -> Any:
+            min_window_index = min([c[0] for c in window_selected])
+            first_wq = Table(f'wq{min_window_index}')
+            merged_query = (
+                self.QUERY_CLS.from_(window_subquery_list[0])
+                .select(
+                    *step.on,
+                    *[
+                        getattr(first_wq, col[1].alias)
+                        for col in window_selected
+                        if col[0] == min_window_index
+                    ],
+                )
+                .as_('window_subquery')
+            )
+            if len(window_subquery_list) > 1:
```

Comment:
I don't think the if guard is needed, since we won't loop anyway if the subquery list has only one element

Reply:
:+1: